lua/rustaceanvim/lsp/init.lua

```diff
@@ -311,12 +311,20 @@ M.set_target_arch = function(bufnr, target)
   restart(bufnr, { exclude_rustc_target = target }, function(client)
     rustc.with_rustc_target_architectures(function(rustc_targets)
       if rustc_targets[target] then
-        client.settings['rust-analyzer'].cargo.target = target
+        -- NOTE: It is not guaranteed that the 'cargo' subkey exists (atleast rustaceanvim does
+        -- not configure it by default).
+        local ra = client.config.settings['rust-analyzer']
+        ra.cargo = ra.cargo or {} -- Initialize the cargo subkey if necessary
```

Comment:
suggestion: The code is clear enough. No need for comments here :smile: 

```suggestion
        local ra = client.config.settings['rust-analyzer']
        ra.cargo = ra.cargo or {}
```